openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -133,6 +89,307 @@ def no_workspace_dir():
+        )
+        extra_args = []
+        for flag, value in extra_args_dict.items():
+            extra_args.append(flag)
+            extra_args.append(value)
+        self.extra_args = extra_args
+
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+        """Process the texture.
+
+        This function requires the `maketx` executable to be available in an
+        OpenImageIO toolset detectable by OpenPype.
+
+        Args:
+            source (str): Path to source file.
+            colorspace (str): Colorspace of the source file.
+            color_management (dict): Maya Color management data from
+                `lib.get_color_management_preferences`
+            staging_dir (str): Output directory to write to.
+
+        Returns:
+            TextureResult: The resulting texture information.
+
+        """
+
+        maketx_path = get_oiio_tools_path("maketx")
+
+        if not maketx_path:
+            raise AssertionError(
+                "OIIO 'maketx' tool not found. Result: {}".format(maketx_path)
+            )
+
+        # Define .tx filepath in staging if source file is not .tx
+        fname, ext = os.path.splitext(os.path.basename(source))
+        if ext == ".tx":
+            # Do nothing if the source file is already a .tx file.
+            return TextureResult(
+                path=source,
+                file_hash=None,     # todo: unknown texture hash?
+                colorspace=colorspace,
+                transfer_mode=COPY
+            )
+
+        args = []
+        if color_management["enabled"]:
+            config_path = color_management["config"]
+            if not os.path.exists(config_path):
+                raise RuntimeError("OCIO config not found at: "
+                                   "{}".format(config_path))
+
+            render_colorspace = color_management["rendering_space"]
+
+            self.log.info("tx: converting colorspace {0} "
+                          "-> {1}".format(colorspace,
+                                          render_colorspace))
+            args.extend(["--colorconvert", colorspace, render_colorspace])
+            args.extend(["--colorconfig", config_path])
+
+        else:
+            # We can't rely on the colorspace attribute when not in color
+            # managed mode because the collected color space is the color space
+            # attribute of the file node which can be any string whatsoever
+            # but only appears disabled in Attribute Editor. We assume we're
+            # always converting to linear/Raw if the source file is assumed to
+            # be sRGB.
+            # TODO Without color management do we even know we can do
+            #      "colorconvert" and what config does that end up using since
+            #       colorconvert is a OCIO command line flag for maketx.
+            #       Also, Raw != linear?
```

Comment:
from https://openimageio.readthedocs.io/en/latest/maketx.html:
> --colorconvert <inspace> <outspace>
Convert the color space of the input image from inspace to tospace. If OpenColorIO is installed and finds a valid configuration, it will be used for the color conversion. If OCIO is not enabled (or cannot find a valid configuration, OIIO will at least be able to convert among linear, sRGB, and Rec709.

seems like we would only be able to do from `sRGB` to `linear` and not `Raw`, I'd change it to:
```
args.extend(["--colorconvert", "sRGB", render_colorspace])
```

Reply:
@moonyuet @antirotor @fabiaserra  What do we want to do in this case?

Previously we'd still rely on using the OCIO config that was set in Maya even if maya color management was disabled. I'd say that's wrong because whatever the OCIO config is then set to in Maya would basically be unused. However, without color management we can't seem to 'linearize' through the maketx.

[This is what Arnold does internally in the tx manager (click on the **"Expand to see script"**)](https://github.com/ynput/OpenPype/pull/4276#discussion_r1071519771) which whenever color management is disabled comes down to only doing this:

```python
        # promote 8-bit images to EXR half with DWAA compression to avoid quantization errors (#3919)
        if colorspace != 'linear' and tile_info['bit_depth'] <= 8:
            cmd += ['--format', 'exr', '-d', 'half', '--compression', 'dwaa']
```

It basically does no color conversion whenever color management is disabled in Maya. Should we be doing the same?

---

Actually, I've decided for now to mimic what Arnold does in tx manager and whenever color management is disabled to do *no* color conversion on `.tx` generation.

Reply:
> @moonyuet @antirotor @fabiaserra What do we want to do in this case?
> 
> Previously we'd still rely on using the OCIO config that was set in Maya even if maya color management was disabled. I'd say that's wrong because whatever the OCIO config is then set to in Maya would basically be unused. However, without color management we can't seem to 'linearize' through the maketx.
> 
> [This is what Arnold does internally in the tx manager (click on the **"Expand to see script"**)](https://github.com/ynput/OpenPype/pull/4276#discussion_r1071519771) which whenever color management is disabled comes down to only doing this:
> 
> ```python
>         # promote 8-bit images to EXR half with DWAA compression to avoid quantization errors (#3919)
>         if colorspace != 'linear' and tile_info['bit_depth'] <= 8:
>             cmd += ['--format', 'exr', '-d', 'half', '--compression', 'dwaa']
> ```
> 
> It basically does no color conversion whenever color management is disabled in Maya. Should we be doing the same?
> 
> Actually, I've decided for now to mimic what Arnold does in tx manager and whenever color management is disabled to do _no_ color conversion on `.tx` generation.

We can still run the tx conversion without color config and color convert, right? I think we should be doing the same - the tx conversion only if there's no color config.  
We need to warn or remind the users the color management disabled and there won't be color conversion when making tx.

Reply:
Thanks @moonyuet that's indeed what I ended up doing.

Reply:
surprise surprise, docs not being accurate haha 

and yeah, good reasoning on replicating Arnold's behavior, my +1 on that